feature/experimental/bgp/ate_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,417 @@
+			WithHoldTime(ateParams.holdTime).WithTypeExternal()
+	}
+	return topo
+}
+
+// TestEstablish sets up and verifies basic BGP connection.
+func TestEstablish(t *testing.T) {
+	// DUT configurations.
+	t.Log("Start DUT config load:")
+	dut := ondatra.DUT(t, "dut")
+
+	// Configure interface on the DUT
+	t.Log("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	// Configure BGP Neighbor on the DUT
+	t.Log("Start DUT Network Instance and BGP Config")
+	configureNetworkInstance(t)
+
+	dutConfPath := dut.Config().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	fptest.LogYgot(t, "DUT BGP Config before", dutConfPath, dutConfPath.Get(t))
+	dutConfPath.Replace(t, nil)
+	dutConf := bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS})
+	dutConfPath.Replace(t, dutConf)
+
+	// ATE Configuration.
+	t.Log("Start ATE Config")
+	topo := configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutAttrs.IPv4, connType: "EXTERNAL"})
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	// Verify Port Status
+	t.Log("Verifying port status")
+	verifyPortsUp(t, dut.Device)
+
+	t.Log("Check BGP parameters")
+	verifyBgpTelemetry(t, dut)
+
+	t.Log("Check BGP Capabilities")
+	verifyBGPCapabilities(t, dut)
+}
+
+// TestDisconnect is send cease notification from ate and to verify reiceved error code on DUT.
+func TestDisconnect(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	// Configure interface on the DUT
+	t.Log("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	dutConfPath := dut.Config().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	statePath := dut.Telemetry().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateAttrs.IPv4)
+
+	// Configure BGP Neighbor on the DUT
+	t.Log("Start DUT Network Instance and BGP Config")
+	configureNetworkInstance(t)
+	// Clear any existing config
+	fptest.LogYgot(t, "DUT BGP Config before", dutConfPath, dutConfPath.Get(t))
+	dutConfPath.Replace(t, nil)
+	dutConf := bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS})
+	dutConfPath.Replace(t, dutConf)
+
+	t.Log("configure port and BGP configs on ATE")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+	iDut1 := topo.AddInterface(ateAttrs.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateAttrs.IPv4CIDR()).WithDefaultGateway(dutAttrs.IPv4)
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dutAttrs.IPv4).WithLocalASN(ateAS).WithTypeExternal()
+
+	t.Log("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	t.Log("Verify BGP session state : ESTABLISHED")
+	nbrPath.SessionState().Await(t, time.Second*100, telemetry.Bgp_Neighbor_SessionState_ESTABLISHED)
+
+	t.Log("Send Cease Notification from ATE to DUT")
+	// TODO: waiting for ATE to support notification.
```

Comment:
ATE support of sending BGP notification has been provided. Please refer : https://github.com/openconfig/ondatra/pull/17